### ISSUE TITLE
Breaking changes in Telegram API v1.6.0

### ISSUE THREAD

**Issue by miqm** [osmandapp/Osmand#8543]
Hi,

I think there's a problem in building OsmAnd-Telegram. There's a step for downloading latest telegram api prebuilt library. But it seems, that there has been a breaking change in latest v1.6 released on 31st January 2020.

When building I'm getting following error message:
```
osmandapp\android\OsmAnd-telegram\src\net\osmand\telegram\helpers\TelegramHelper.kt: (444, 24): None of the following functions can be called with the arguments supplied:
public constructor GetChats() defined in org.drinkless.td.libcore.telegram.TdApi.GetChats
public constructor GetChats(chatList: TdApi.ChatList!, offsetOrder: Long, offsetChatId: Long, limit: Int) defined in org.drinkless.td.libcore.telegram.TdApi.GetChats
```
I see in history, that this problem appeared in the past as well (commits "compilation fix").

I'd suggest storing tdlib.zip in OsmAnd repository, to avoid this issue in future, especially when telegram does not provide ability to download specific versions (which is quite odd approach).

**Comment by vshcherb** [comment]
Yes it happens but we are not going to store it in this repository, probably we will cache in our ivy repository.
You are right we need to update and get it fixed

**Comment by vshcherb** [comment]
As of now it is always cached with our build here https://builder.osmand.net:8080/view/OsmAnd%20Builds/job/OsmAnd-tracker/ws/android/OsmAnd-telegram/tdlib.zip
